### PR TITLE
Limit trashing

### PR DESCRIPTION
**Pull request by appgurueu**
Items now need trashable > 0 to be trashable, class items remain trashable as this is the only way to get rid of them.

If you need the inventory space, drop your items.

What this PR does:

* Unregisters `/clearinv` and `/pulverize`
* Adds a trash slot to the team chest pro section, removes the trash slot from your inventory
* Only allows trashing items with `trashable > 0` (currently only stone tools)

### CODE REVIEW

**Comment by appgurueu** [comment]
I should also remove `/clearinv` but I feel this won't go anywhere.

**Comment by appgurueu** [comment]
Removing trashing from the inventory is problematic because crafting currently is a submodule.

**Comment by LoneWolfHT** [comment]
Don't remove it, leave it for all but the most powerful items

File: mods/ctf/ctf_map/ctf_map_core/chest.lua [hunk 1]
```diff
@@ -40,11 +41,12 @@ for _, chest_color in pairs(colors) do
 	}
 
 	function def.on_construct(pos)
+		ctf_map.chest_locations[chest_color] = pos
 		local meta = minetest.get_meta(pos)
 		meta:set_string("infotext", "Chest")
 		local inv = meta:get_inventory()
 		inv:set_size("main", 4 * 7)
-		inv:set_size("pro", 4 * 7)
+		inv:set_size("pro", 4 * 7 - 1) -- one slot left for trash
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_map/ctf_map_core/chest.lua:49]: This doesn't sound good

File: mods/ctf/ctf_trashlimit/init.lua [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	if not ctf_stats.is_pro(name) then
+		return 0
+	end
+	local itemname = stack:get_name()
+	if minetest.get_item_group(itemname, "trashable") == 0 and not ctf_classes.item_is_class_blacklisted(player, itemname) then
+		minetest.chat_send_player(name, ("%s is not trashable!"):format(stack:get_description()))
+		return 0
+	end
+	return allow_put(inv, listname, index, stack, player)
+end
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_trashlimit/init.lua:22]: No newline at the end of the file

File: mods/ctf/ctf_trashlimit/mod.conf [hunk 3]
```diff
@@ -0,0 +1,3 @@
+name = ctf_trashlimit
+depends = crafting
+description = Limits trashing to items with the "trashable" group
```
Comment (LoneWolfHT) [inline mods/ctf/ctf_trashlimit/mod.conf:3]: No newline here either

File: mods/ctf/ctf_map/ctf_map_core/chest.lua (hunk 1 above)
Comment (appgurueu) [reply mods/ctf/ctf_map/ctf_map_core/chest.lua:49]: Why?

Comment (LoneWolfHT) [reply mods/ctf/ctf_map/ctf_map_core/chest.lua:49]: Not sure if it'll be visually separate enough

**Comment by LoneWolfHT** [comment]
> > Don't remove it, leave it for all but the most powerful items
> 
> Should be the other way around, otherwise entire stacks of ammo/medkits/whatever can be trashed (which could be seen as borderline sabotage BTW).

I'm ok with moving the trashing of powerful weapons to the team chest. But any more than that I'm no longer supportive of.

**Review by Avyukt1 — APPROVED** [review]

File: mods/ctf/ctf_stats/init.lua [hunk 4]
```diff
@@ -142,7 +142,11 @@ function ctf_stats.player(name)
 
 	local match_player_stats =
 			ctf_stats.current.red[name] or ctf_stats.current.blue[name]
-
+	if name == "singleplayer" then
```
Comment (appgurueu) [inline mods/ctf/ctf_stats/init.lua:145]: This testing code will be removed for merge.

**Comment by appgurueu** [comment]
Bump. Trashing still needs to be fixed.

**Comment by appgurueu** [comment]
Reason for close?

**Comment by savilli** [comment]
We have decided we don't need that

**Pull request closed without merging**